### PR TITLE
refactor(widget-space): Move error handler into enhancer

### PR DESCRIPTION
**Pull request by bzang**

### CODE REVIEW

File: packages/node_modules/@ciscospark/widget-space/src/container.js [hunk 1]
```diff
@@ -417,18 +363,7 @@ export class SpaceWidget extends Component {
...
-              <ErrorDisplay
-                actionTitle={errorActionTitle}
-                onAction={errorAction}
-                secondaryTitle={displaySubtitle}
-                title={displayTitle}
-                transparent={temporary}
-              />
-            </div>
-          }
+          { errors.get(`hasError`) && errorElement }
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/widget-space/src/container.js]: since we aren't populating `errorElement` above unless `hasError` you could probably just put `{ errorElement }` in

File: packages/node_modules/@ciscospark/widget-space/src/enhancers/errors.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+export default compose(
+  connect(
+    (state) => state,
+    (dispatch) => bindActionCreators({
+      addError,
+      removeError
+    }, dispatch)
+  ),
+  lifecycle({
+    componentWillMount() {
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/widget-space/src/enhancers/errors.js]: This is really cool

File: packages/node_modules/@ciscospark/widget-space/src/index.html [hunk 3]
```diff
@@ -34,10 +34,10 @@
     <main
       class="ciscospark-space"
       id="ciscospark-widget-direct-data-api"
-      data-initial-activity="meet"
+      data-initial-activity="message"
       data-toggle="ciscospark-space"
       data-access-token="<%= process.env.CISCOSPARK_ACCESS_TOKEN %>"
-      data-to-person-email="<%= process.env.TO_PERSON_EMAIL %>"
+      data-to-person-email="nzang@cisco.com"
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/widget-space/src/index.html]: Probably want to revert these changes :)

File: packages/node_modules/@ciscospark/widget-space/src/container.js (hunk 1 above)
Comment (bzang) [reply packages/node_modules/@ciscospark/widget-space/src/container.js]: Good call

File: packages/node_modules/@ciscospark/widget-space/src/enhancers/errors.js (hunk 2 above)
Comment (bzang) [reply packages/node_modules/@ciscospark/widget-space/src/enhancers/errors.js]: Yea. This is just one of the utilities. There a a bunch of others that simply patterns we already use. It also has a few that allow us to take base presentational components and apply transforms to make them custom components that we can reuse easily

File: packages/node_modules/@ciscospark/widget-space/src/index.html (hunk 3 above)
Comment (bzang) [reply packages/node_modules/@ciscospark/widget-space/src/index.html]: Haha... yea. Gonna force push this one away

**Comment by bzang** [comment]
Ready for another round @adamweeks

File: packages/node_modules/@ciscospark/widget-space/src/container.js [hunk 4]
```diff
@@ -177,36 +176,19 @@ export class SpaceWidget extends Component {
...
-        id: registerErrorId,
-        displayTitle: formatMessage(messages.unableToLoad),
-        displaySubtitle,
-        temporary: false,
-        code: error.statusCode
-      });
-    }
 
-    else if (sparkInstance && sparkState.get(`authenticated`) && sparkState.get(`registered`) && !sparkState.get(`hasError`)) {
+    if (sparkInstance
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/widget-space/src/container.js]: Wonder if we should make a helper field for spark state that does these checks for us. Like `sparkState.status === 'ready'`

Comment (bzang) [reply packages/node_modules/@ciscospark/widget-space/src/container.js]: Yea, I'm gonna try to integrate that into the refactor I'm doing.
